pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -199,21 +199,21 @@ func (c *Cluster) Start() error {
 	updateConfig := c.newUpdateConfig(config, updateQueue, deployments)
 
 	// prepare for creating new OSDs
-	statusConfigMaps := util.NewSet()
+	statusConfigMaps := sets.NewString()
 
 	logger.Info("start provisioning the OSDs on PVCs, if needed")
 	pvcConfigMaps, err := c.startProvisioningOverPVCs(config, errs)
 	if err != nil {
 		return err
 	}
-	statusConfigMaps.AddSet(pvcConfigMaps)
+	statusConfigMaps.Union(pvcConfigMaps)
 
 	logger.Info("start provisioning the OSDs on nodes, if needed")
 	nodeConfigMaps, err := c.startProvisioningOverNodes(config, errs)
 	if err != nil {
 		return err
 	}
-	statusConfigMaps.AddSet(nodeConfigMaps.Copy())
+	statusConfigMaps.Union(nodeConfigMaps)
```

Comment:
I got the mistake, Union returns a different set which I need to store, instead of appending the list to the first instance.:)